spec/catalog.yaml

```diff
@@ -59,4 +59,205 @@ paths:
+        type:
+          type: string
+          description: Internal type definition of the error
+          example: NoSuchDatabaseException
+        code:
+          type: integer
+          minimum: 400
+          maximum: 600
+          description: HTTP response code
+          example: 404
+        stack:
+          type: array
+          items:
+            type: string
+
+    LanceCatalogErrorResponse:
+      description: JSON wrapper for all error responses (non-2xx)
+      type: object
+      required:
+        - error
+      properties:
+        error:
+          $ref: '#/components/schemas/ErrorModel'
+      additionalProperties: false
+      example:
+        {
+          "error": {
+            "message": "The server does not support this operation",
+            "type": "UnsupportedOperationException",
+            "code": 406
+          }
+        }
+
+    CreateDatabaseRequest:
+      type: object
+      required: [ name, mode ]
+      properties:
+        name: { type: string }
+        mode:
+          type: string
+          enum: [ CREATE, EXIST_OK, OVERWRITE ]
+        options:
+          type: object
+          additionalProperties: { type: string }
+
+    CreateDatabaseResponse:
+      type: object
+      required:
+        - database
+      properties:
+        database: { type: string }
+        properties:
+          type: object
+          additionalProperties:
+            type: string
+          description:
+            Properties stored on the database, if supported by the server.
+          example: { "created_at": "1452120468" }
+          default: { }
+
+  responses:
+    CreateDatabaseResponse:
+      description:
+        Represents a successful call to create a database.
+        Returns the database created, as well as any properties that were stored for the database,
+        including those the server might have added. Implementations are not required to support database
+        properties.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/CreateDatabaseResponse'
+          example: {
+            "database": [ "lance_database" ],
+            "properties": { "created_at": "1452120468" }
+          }
+
+    BadRequestErrorResponse:
+      description:
+        Indicates a bad request error. It could be caused by an unexpected request
+        body format or other forms of request validation failure, such as invalid json.
+        Usually serves application/json content, although in some cases simple text/plain content might
+        be returned by the server's middleware.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/LanceCatalogErrorResponse'
+          example: {
+            "error": {
+              "message": "Malformed request",
+              "type": "BadRequestException",
+              "code": 400
+            }
+          }
+
+    ForbiddenResponse:
```

Comment:
I think we should also add 401 error response 

https://beeceptor.com/docs/concepts/401-vs-403/#:~:text=401%20indicates%20missing%20or%20invalid,of%20permission%20despite%20valid%20authentication.

Reply:
What about adding it when we systematic consideration of permission?

Reply:
sure!